engine/runtime/src/test/scala/org/enso/interpreter/test/semantic/ErrorsTest.scala

```diff
@@ -133,5 +133,27 @@ class ErrorsTest extends InterpreterTest {
       val code = "main = 10.catch (x -> x + 1)"
       eval(code) shouldEqual 10
     }
+
+    "catch polyglot errors" in {
+      val code =
+        """from Builtins import all
+          |polyglot java import java.lang.Long
+          |
+          |main =
+          |    args = Array.new 1
+          |    args.set_at 0 "oops"
+          |    caught = Panic.recover (Long.parseLong args)
+          |    IO.println caught
+          |    unwrap = caught.catch <| case _ of
+          |        Polyglot_Error err -> err
+          |        _ -> "fail"
+          |    IO.println unwrap
```

Comment:
Please verify that `unwrap` is a legit polyglot object – try calling some method on it and see if the result is still usable.